src/pages/on-chain-governance/_components/ConfirmVoteDialog.tsx

```diff
@@ -171,166 +159,249 @@ export function ConfirmVoteDialog({
+            isMasterNodeInputFocus ? "border-primary-300" : "border-gray-300 ",
+            { "border-red-500": masterNodeErrorMsg !== "" }
+          )}
+        >
+          <textarea
+            ref={ref}
+            onChange={(v) => {
+              if (v.target.value.length !== 64) {
+                setMasterNodeErrorMsg("Please enter a valid masternode ID");
+              } else {
+                setMasterNodeErrorMsg("");
+              }
+              setMasterNodeID(v.target.value);
             }}
-            className={classNames(
-              "flex flex-row rounded border py-3 px-4 w-full dark:bg-gray-800",
-              { "border-primary-300": isMasterNodeInputFocus },
-              { "border-red-600": masterNodeErrorMsg !== "" }
-            )}
-          >
-            <textarea
-              onChange={(v) => {
-                if (v.target.value.length !== 64) {
-                  setMasterNodeErrorMsg("Enter a valid masternode ID");
-                } else {
-                  setMasterNodeErrorMsg("");
-                }
-                setMasterNodeID(v.target.value);
-              }}
-              disabled={!isMasterNodeEditClicked}
-              value={masterNodeID}
-              className="overflow-hidden resize-none md:h-5 h-10 text-sm focus:outline-none grow focus:caret-[#007AFF] dark:bg-gray-800 dark:text-dark-gray-900 disabled:bg-white dark:disabled:bg-gray-800"
-              placeholder="Set your masternode"
-            />
-
-            {(masterNodeID !== "" || isMasterNodeInputFocus) &&
-              isMasterNodeEditClicked && (
-                <MdClear
-                  onClick={() => {
-                    setMasterNodeID("");
-                  }}
-                  size={15}
-                  className="text-gray-500 self-center cursor-pointer ml-[6px]"
-                />
-              )}
-            {!isMasterNodeEditClicked && (
-              <MdEdit
-                role="button"
-                size={18}
-                onClick={() => setIsMasterNodeEditClicked(true)}
-                className="text-primary-500 ml-[6px] self-center"
-              />
-            )}
-          </div>
+            value={masterNodeID}
+            className="overflow-visible resize-none text-sm focus:outline-none focus:caret-[#007AFF] dark:bg-gray-800 text-gray-900 dark:text-dark-gray-900 disabled:bg-white dark:disabled:bg-gray-800 placeholder:text-gray-400 grow tracking-[0.0025em]"
+            placeholder="Masternode ID"
+          />
 
-          {isMasterNodeEditClicked && (
-            <>
-              <Button
-                label="SAVE"
-                testId="OnChainGovernance.SaveMasterNodeID"
-                disabled={masterNodeID === "" || masterNodeErrorMsg !== ""}
-                onClick={() => {
-                  setIsMasterNodeEditClicked(false);
-                  localStorage.setItem("masternodeID", masterNodeID);
-                }}
+          {isMasterNodeInputFocus && (
```

Comment:
close button should display when there is input, regardless if input is in autofocus or not

Reply:
updated to fix width of the input text and use justify so that text would not jump
<img width="444" alt="image" src="https://user-images.githubusercontent.com/57183851/211491619-aeb7f8a2-d239-432b-a80f-9a2a13ed6c06.png">